### PR TITLE
DDCE-2640 : Update libs + add scalafmt + fix scalastylye warns

### PR DESCRIPTION
**Pull request by lmugnai**
# DDCE-(StoryNumberHere)

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date
 
## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Review by frikit — CHANGES_REQUESTED** [review]
- Please update uk.gov.hmrc:bootstrap-backend-play-28 : 7.1.0 -> 7.2.0
- Please update uk.gov.hmrc:tax-year : 1.4.0 -> 1.7.0 -> 3.0.0

If you could fix bellow scalastyle warnings that would be great:
`
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/model/api/Employment.scala:42:6: Public method must have explicit type
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/model/api/Employment.scala:44:6: Public method must have explicit type
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/controllers/EmploymentDetailController.scala:65: File line length exceeds 160 characters
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/controllers/BaseController.scala:64:16: Cyclomatic complexity of 11 exceeds max of 10
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/controllers/SelectTaxYearController.scala:97:8: Regular expression matched 'println'
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/controllers/SelectTaxYearController.scala:100:8: Regular expression matched 'println'
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/controllers/SelectTaxYearController.scala:118:14: Regular expression matched 'println'
[warn] /Users/frikit/IdeaProjects/tax-history-frontend/app/controllers/SelectTaxYearController.scala:121:14: Regular expression matched 'println'
`

**Comment by lmugnai** [comment]
@frikit all comments addressed.

**Comment by lmugnai** [comment]
@frikit Addressed all comments

**Review by frikit — CHANGES_REQUESTED** [review]

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -7,30 +7,29 @@ object AppDependencies {
...
-    "com.typesafe.play"      %% "play-json-joda"                   % "2.9.2"
-
+    "uk.gov.hmrc"       %% "domain"                     % "8.1.0-play-28",
+    "uk.gov.hmrc"       %% "bootstrap-frontend-play-28" % "7.2.0",
+    "uk.gov.hmrc"       %% "play-frontend-hmrc"         % "3.24.0-play-28",
+    "uk.gov.hmrc"       %% "play-partials"              % "8.3.0-play-28",
+    "uk.gov.hmrc"       %% "url-builder"                % "3.6.0-play-28",
+    "uk.gov.hmrc"       %% "agent-mtd-identifiers"      % "0.47.0-play-28",
+    "uk.gov.hmrc"       %% "tax-year"                   % "3.0.0",
+    "com.typesafe.play" %% "play-json-joda"             % "2.9.3"
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply project/AppDependencies.scala]: done

**Review by frikit — APPROVED** [review]

**Pull request merged**